### DISCUSSION TITLE
which data buffer is filled when the host sends data to device over USB HID in TinyUSB?

### DISCUSSION THREAD

**Discussion by nachiketathakur** [hathach/tinyusb#1399] [Q&A]
I am developing on RP2040(Pico Development board) and in my application, I want to receive data from HOST. What I cannot find is which buffer is filled and how when the Host sends any data? 
EDIT: I am referring the dev_hid_composite example.

**Comment by hathach** [comment]
try hid_generic_inout, there is a python + node script that send and receive data from host to device

**Reply by nachiketathakur** [reply]
I tried example, but I am facing the issue https://github.com/raspberrypi/pico-examples/issues/93#issue-836964895 .

**Reply by hathach** [reply]
It is not clear to me what you have done and what is the issue. Would you mind providing steps to reproduce the issue (take a look at other issue), along with debug log (LOG=2) from device and output from python host.

Note: if you need to clone and build with this repo as opposed to one included in pico-sdk
